### ISSUE TITLE
Library Compilation Error (0.7.0): conflicting implementations of trait `error::Error<_>` for type `error::Simple<_, _>`

### ISSUE THREAD

**Issue by ColinTimBarndt** [zesterer/chumsky#72]
I am not sure why, but Rust cannot compile the library version 0.7.0. The error can be seen below. Could someone else please also confirm this?

<details>

```txt
error[E0119]: conflicting implementations of trait `error::Error<_>` for type `error::Simple<_, _>`
   --> /home/colin/.cargo/registry/src/github.com-1ecc6299db9ec823/chumsky-0.7.0/src/error.rs:352:1
    |
223 | impl<I: Hash + Eq, S: Span + Clone + fmt::Debug> Error<I> for Simple<I, S> {
    | -------------------------------------------------------------------------- first implementation here
...
352 | impl<I, S: Span + Clone + fmt::Debug> Error<I> for Simple<I, S> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `error::Simple<_, _>`

For more information about this error, try `rustc --explain E0119`.
error: could not compile `chumsky` due to previous error
```

</details>

In the current versions, the error corresponds to [here](https://github.com/zesterer/chumsky/blob/60cba221fbc625eecd1f77bd4ae76642181e5297/src/error.rs#L231) and [here](https://github.com/zesterer/chumsky/blob/60cba221fbc625eecd1f77bd4ae76642181e5297/src/error.rs#L360) in the current source code.

Notice that in the latter `Simple` was changed to `Cheap` at some point, though I couldn't find out when.

**Comment by ColinTimBarndt** [comment]
A workaround I am currently using is to clone the repository locally and specify

```toml
[patch.crates-io]
chumsky = { path = "./chumsky" }
```

This compiles, so the next stable release will definitely fix this issue.

**Comment by zesterer** [comment]
Hmm, interesting. Perhaps this is a typo? Regardless, as you say, it shouldn't be present in the next release that should be coming in the next day or two.

**Comment by zesterer** [comment]
Just an FYI, I've released version `0.8` now so this should no longer be can issue. If the issue persists, I can re-open this.
